### ISSUE TITLE
Allow setting default shell via profiles at build

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#1919]
To assist distros with non-default shell locations like nixos, we should allow profiles to define a default shell override. 

We also should improve the errors when a shell isn't found to direct users to inspect /etc/shells.

**Comment by yaleman** [comment]
This feels like something that should be a config option, based on org instead of distribution?

**Comment by Firstyear** [comment]
we can also set it via the unixd config. So there is already a hierachy of precedence which is:

* user defined shell
* config (org) defined default shell
* distro defined default shell.

**Comment by yaleman** [comment]
Sounds awesome, thanks for the underp-reminder <3

**Comment by Firstyear** [comment]
Yeah, effectively what failed here was that all three values were unset or invalid, and so it fell through to the "hard to parse" error case. So we need to make that error better *and* give a better tool for distros to prevent the third case falling through.
